Makefile

```
@@ -1,24 +1,55 @@
-.PHONY: all clean install uninstall distclean
+.PHONY: all clean install uninstall distclean ocaml
 
 include Makeconf
 
-FREESTANDING_LIBS=openlibm/libopenlibm.a \
-		  ocaml/runtime/libasmrun.a \
-		  nolibc/libnolibc.a
+all:	openlibm/libopenlibm.a nolibc/libnolibc.a ocaml ocaml-freestanding.pc freestanding.conf
 
-all:	$(FREESTANDING_LIBS) ocaml-freestanding.pc
+TOP=$(abspath .)
 
-Makeconf:
-	./configure.sh
+# CFLAGS used to build nolibc / openlibm / ocaml runtime
+LOCAL_CFLAGS=$(MAKECONF_CFLAGS) -I$(TOP)/nolibc/include -include _freestanding/overrides.h
+# CFLAGS used by the OCaml compiler to build C stubs
+GLOBAL_CFLAGS=$(MAKECONF_CFLAGS) -I$(MAKECONF_PREFIX)/freestanding-sysroot/include/nolibc/ -include _freestanding/overrides.h
+# LIBS used by the OCaml compiler to link executables
+GLOBAL_LIBS=-L$(MAKECONF_PREFIX)/freestanding-sysroot/lib/nolibc/ -lnolibc -lopenlibm $(MAKECONF_EXTRA_LIBS)
 
-TOP=$(abspath .)
-FREESTANDING_CFLAGS+=-I$(TOP)/nolibc/include -include _freestanding/overrides.h
+# NOLIBC
+NOLIBC_CFLAGS=$(LOCAL_CFLAGS) -I$(TOP)/openlibm/src -I$(TOP)/openlibm/include
+nolibc/libnolibc.a:
+	$(MAKE) -C nolibc \
+	    "CC=$(MAKECONF_CC)" \
+	    "FREESTANDING_CFLAGS=$(NOLIBC_CFLAGS)" \
+	    "SYSDEP_OBJS=$(MAKECONF_NOLIBC_SYSDEP_OBJS)"
 
+# OPENLIBM
 openlibm/libopenlibm.a:
-	$(MAKE) -C openlibm "CFLAGS=$(FREESTANDING_CFLAGS)" libopenlibm.a
+	$(MAKE) -C openlibm "CC=$(MAKECONF_CC)" "CPPFLAGS=$(LOCAL_CFLAGS)" libopenlibm.a
 
+# OCAML
 ocaml/Makefile:
 	cp -r `ocamlfind query ocaml-src` ./ocaml
```

```suggestion
	cp -r `ocamlfind query ocaml-src` ./ocaml
#configure avoid inserting -lm
	sed -i -e 's/ac_cv_lib_m_cos=yes/ac_cv_lib_m_cos=no/' ocaml/configure
```

as learned by #101 // #102

---

alternatively, we can pass `ac_cv_lib_m_cos=no` to configure below (not tested)